Include the info on different environments (#2647)
thoth-station/thoth-application: docs/environments.md

# Deployment Environments

This section describes the different environments used for all the Thoth Services deployments on the OpenShift Cluster.

*Note*: All environments can be deployed using ArgoCD or Kustomize.

> ## Prod Environments

Reviewer: ```suggestion
## Prod Environment
```
Might use the singular ? AFAICT it's one env right ?

The production environment is deployed on the AWS cluster named **balrog** hosted by Operate-First.<br>
The access to this cluster is only available for [thoth-dev-ops]() team only.<br>
The deployment on this cluster is set via ArgoCD and also set on self-healing and pruning resources.<br>
It uses an ephemeral PostgreSQL deployment, Kafka Broker deployment.

Server: [balrog-opf-aws](https://console-openshift-console.apps.balrog.aws.operate-first.cloud/)<br>
Request access: Open PR to be included in [list of users](https://github.com/operate-first/apps/blob/master/cluster-scope/base/user.openshift.io/groups/thoth-devops/group.yaml)

> ## Semi-Prod Environments

Reviewer: Same as above

> This environment is used as a temperory production, it is on the same standard as production in term of resources,

Reviewer: ```suggestion
This environment is used as a temporary production, it is on the same standard as production in term of resources,
```

the performance might be affected.<br>
The semi-production environment is deployed on the MOC cluster named **smaug** hosted by Operate-First.<br>
The access to this cluster is only available for [thoth-dev-ops]() team only.<br>
The deployment on this cluster is set via ArgoCD and also set on self-healing and pruning of resources.<br>
It uses an ephemeral PostgreSQL deployment,and depends on an external Kafka Broker deployment.

Server: [smaug-opf-moc](https://console-openshift-console.apps.smaug.na.operate-first.cloud/)<br>
Request access: Open PR to be included in [list of users](https://github.com/operate-first/apps/blob/master/cluster-scope/base/user.openshift.io/groups/thoth-devops/group.yaml)

> ## Stage Environments

Reviewer: Same as above

This environment is behind **Red Hat VPN**, it would require VPN for access.<br>
The stage environment is deployed on the OCP cluster named **ocp4** hosted by PSI.<br>
The access to this cluster is only available for [thoth](https://rover.redhat.com/groups/group/aicoe-thoth) team only,which is configured directly via ldap groups.<br>
The deployment on this cluster is set via ArgoCD and also set on self-healing and pruning of resources.<br>
It uses an ephemeral PostgreSQL deployment,and depends on an external Kafka Broker deployment.

Server: [ocp4-psi](https://ocp4.psi.redhat.com/)<br>
Request access: Contact the Manager to be included in the [group](https://rover.redhat.com/groups/group/aicoe-thoth).

## Test Environments

> The Test environments are volatail and can be multiple at a give point of time, however devs can deploy there own if needed on the quicklab cluster (read here for more). The hosted test environment by team can be broken at any time, and ArgoCD is not set on self-healing or pruning of resources.<br> The test environment consists of one OpenShift project, uses an ephemeral PostgreSQL deployment inside of that project, and depends on an external Kafka Broker deployment.<br>

Reviewer: ```suggestion
The Test environments are volatile and can be multiple at any given point in time, however devs can deploy their own if needed on the quicklab cluster (read here for more). The hosted test environment by team can be broken at any time, and ArgoCD is not set on self-healing or pruning of resources.<br>
The test environment consists of one OpenShift project, uses an ephemeral PostgreSQL deployment inside of that project, and depends on an external Kafka Broker deployment.<br>
```
spell.

Reviewer: Also, what exactly does it mean that test environments can be multiple at any given in time ? Shouldn't we simply document what exists ?

## Test Environments

The Test environments are volatile and can be multiple at any given point in time, however devs can deploy their own if needed on the quicklab cluster (read here for more). The hosted test environment by team can be broken at any time, and ArgoCD is not set on self-healing or pruning of resources.<br>
The test environment consists of one OpenShift project, uses an ephemeral PostgreSQL deployment inside of that project, and depends on an external Kafka Broker deployment.<br>
The test environment belonging to the team is used to build Pull Requests.

Standard one:
The hosted environment is behind **Red Hat VPN**, it would require VPN for access.<br>
It is deployed on the OCP cluster named **ocp4** hosted by PSI.<br>
The access to this cluster is only available for [thoth](https://rover.redhat.com/groups/group/aicoe-thoth) team only,which is configured directly via ldap groups.

Server: [ocp4-psi](https://ocp4.psi.redhat.com/)<br>
Namespace: **thoth-test-core**<br>
Request access: Contact the Manager to be included in the [group](https://rover.redhat.com/groups/group/aicoe-thoth).

> 

Reviewer: Not sure if several "Credentials" section would conflict though (on the anchor level I mean)

------------------- | --------- | ---------
inspection-build    | 1         | 1Gi
inspection-run      | 2.5       | 2Gi

## Database Setup

Project Thoth uses Postgresql database for knowlegde graph storage.

- Deployment manifest can be found at: [Graph Manifest](../core/overlays/moc-prod/graph-prod)

> - we manage the storage via different project Thoth components which utlises the [thoth-storage](https://pypi.org/project/thoth-storages/) module. [Details](https://github.com/thoth-station/storages#thoth-storages)

Reviewer: ```suggestion
- we manage the storage via different project Thoth components which utlises the [thoth-storages](https://pypi.org/project/thoth-storages/) module. [Details](https://github.com/thoth-station/storages#thoth-storages)
```
